include/circt/Dialect/StaticLogic/StaticLogic.td

```diff
@@ -0,0 +1,126 @@
+//===- StaticLogic.td - StaticLogic Definitions ------------*- tablegen -*-===//
+//
+// Copyright 2020 The CIRCT Authors.
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//===----------------------------------------------------------------------===//
+
+#ifdef STATICLOGIC_OPS
+#else
+#define STATICLOGIC_OPS
+
+#ifdef OP_BASE
+#else
+include "mlir/IR/OpBase.td"
+#endif // OP_BASE
+
+include "mlir/IR/SymbolInterfaces.td"
+include "mlir/IR/RegionKindInterface.td"
+include "mlir/Interfaces/CallInterfaces.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+def StaticLogic_Dialect : Dialect {
+  let name = "staticlogic";
+  let cppNamespace = "";
+}
+
+def PipelineOp : Op<StaticLogic_Dialect, "pipeline", [FunctionLike, Symbol,
```

Comment:
Why FunctionLike?  I was thinking that this was just a region inside the op.

Reply:
My thinking was all kinds of statically scheduled modules are represented as separate function-like regions. I don't have a clear sense when function-like region should / shouldn't be used?

Reply:
I see. My thinking was all kinds of statically scheduled operations should be separate modules. But actually this isolation is not necessary to happen until further lowering these operations down to FIRRTL or etc. Before that, the pipeline operation is just kind of annotation and keeping things simple will make it easier to modify the pipeline structure.

Reply:
Even when lowering to FIRRTL, making them separate modules seems to be optional as well.